pkg/ring/kv/memberlist/memberlist_client.go

```diff
@@ -475,6 +484,35 @@ func (m *KV) joinMembersOnStartup(ctx context.Context, members []string) error {
 	return lastErr
 }
 
+// Provides a dns-based member disovery to join a memberlist cluster w/o knowning members' addresses upfront.
+func (m *KV) discoverMembers() {
```

Comment:
I would prefer to not manipulate the config. I would suggest to pick in input `m.cfg.JoinMembers` and return a list of addresses and then passing these addresses to the `m.joinMembersOnStartup()` and `m.memberlist.Join()` instead of `m.cfg.JoinMembers`.